foundations/javascript_basics/revisiting_rock_paper_scissors.md

```diff
@@ -1,17 +1,50 @@
+
+Branches in Git allow your repository to hold multiple *alternate reality* versions of your files at the same time. You’ve actually (sort of) been using branches since you made your first commit, you just might not have known it! Back in [the setting up Git lesson](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/setting-up-git) when you ran `git config --global init.defaultBranch main` you were setting the name of what’s called the *default* branch for your repos. The default branch is just what we call the branch that is created when you make your first commit on a project, and in that command we set the name to be `main` as is the current standard. 
+
+Like the branches in a tree (hence the name) all of the branches for a project stem off of a “trunk” (the `main` branch) or off of *other* branches.
+
+When you make commits on a specific branch those changes only exist on **that** branch, leaving all of your other branches exactly as they were when you branched off of them.
+
+This means that you can keep your `main` branch as a place for only finished features, that you know are working properly, and add each feature to your project using dedicated branches which we call *feature branches*
+
+### Using Branches
+
+You can make new branches by using the command `git branch <branch_name>`. You can then switch to your new branch using `git switch <branch_name>`. You can also create a new branch and switch to it in a single command by using the `-c` (as in “create”) flag with `switch`, in the form `git switch -c <branch_name>`.
+
+You can see all of your current branches using `git branch` with no other arguments. The branch that you’re currently on will be indicated with an asterisk. If you want to change back to `main` from any other branch, you can do so just like switching to any other branch using `git switch main`.
+
+Once you’re done working on your feature branch and you’re ready to bring the commits that you’ve made on it to your main branch,  you need to perform what is known as a `merge`.
+
+Merges are done by using the command `git merge <branch_name>` which will take the changes you’ve committed in `branch_name` and add them to the branch that you’re currently on.
+
+Sometimes the same lines in a file will have been changed by two different branches, when this happens you will have a *merge conflict* when you try and merge those branches together. In order to finish merging the branches, you will have to first resolve the conflict, but we will go into detail about how to do so in a future lesson.
+
+When you don’t need a branch anymore it can be deleted using `git branch -d <branch_name> if the branch has already been merged into `main` or with `git branch -D <branch_name>` if it hasn’t. You will usually want to delete branches when you’re done with them, otherwise they can pile up and make it more difficult to find the branch you’re looking for when you need it.
```

Comment:
First sentence missing a ` after `git branch -d <branch_name> . So it should be "When you don’t need a branch anymore it can be deleted using `git branch -d <branch_name>` if the branch has already been merged into `main` or"